src/decoder/stream.rs

```diff
@@ -306,6 +318,72 @@ impl EndianReader for LZWReader {
     }
 }
 
+///
+/// ## JPEG Reader (for "new-style" JPEG format (TIFF compression tag 7))
+///
+
+pub struct JpegReader {
+    buffer: io::Cursor<Vec<u8>>,
+    byte_order: ByteOrder, // for jpeg image data always ...,
+}
```

Comment:
I'd rather we keep this `pub(crate)` for now. I'm not sure if we want to commit to supporting this exact interface for the long term, also due to some efficiency concerns.